tests/test_set_config.py

```diff
@@ -228,3 +228,23 @@ def test_incompatible_file_type():
         outcome = "Set config failure: bad file type\n"
 
         assert result == outcome
+
+def test_set_config_dry_run():
+    ''' Test that providing a YAML base file with a dry run flag 
+    will print an YAML config file'''
+
+    input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/fruit_config.yaml"))
+
+    args = ['-i', input_file, '-d']
+
+    outstring = io.StringIO()
+    with redirect_stdout(outstring):
+        config.YAMLConfig(input_file)
+    expected = outstring.getvalue()
```

Comment:
I think the issue with the test failure is somewhere in here. When you create this object, you don't really get any stdout information.

Would the test work if expected is defined simply like this?

```
expected = config.YAMLConfig(input_file)
```

Reply:
I have this assertion error when I try that:
assert '{"fruit": "papaya", "vegetable": "eggplant", "how_many": 17, "dressing": "ranch"}\n' == {"fruit": "papaya", "vegetable": "eggplant", "how_many": 17, "dressing": "ranch"}

Reply:
Interesting...can you strip the new line from `expected`?

Reply:
Ok - stripping it from result and converting expected to a string worked.